unit_tests/CServerConfiguration_tests.cpp

```diff
@@ -1,32 +1,41 @@
 #define BOOST_TEST_MODULE "CServerConfiguration Unit Tests"
-#include <boost/test/unit_test.hpp>
-#include <cstring>
-#include <fstream>
-#include <cstdarg>
 #include "config.h"
 #include "db.h"
 #include "diku_exception.h"
+#include "utility.h"
+
+#include <cstdarg>
+#include <cstring>
+#include <fstream>
+
+#include <boost/test/unit_test.hpp>
 
 struct CServerConfiguration_Fixture
 {
-    static CServerConfiguration_Fixture *m_fixptr;
     static const std::string server_config;
     CServerConfiguration_Fixture()
     {
-        // Get temp filename for config file for tests
-        std::tmpnam(fake_server_config_filename);
+        // No way to suppress the warning from tempnam because its hardcoded into lib
+        // so we'll do it here it's only a unit test
+        srand(time(nullptr));
+        std::ostringstream sstrm;
+        sstrm << "tmp" << std::setfill('0') << std::setw(10) << rand();
+        fake_server_config_filename = sstrm.str();
+
         std::ofstream strm(fake_server_config_filename);
         strm << server_config;
-    }
 
-    char fake_server_config_filename[L_tmpnam]{};
+        // Redirect away application logging to tmp file
+        g_log_file_fd = tmpfile();
```

Comment:
Because slog has been redone the hiding of messages with a dummy func doesn't work - so redirect them to a temporary file. The file is automatically deleted by tmpfile() itself